packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -123,6 +123,12 @@ class _SelectableTextSelectionGestureDetectorBuilder extends TextSelectionGestur
 /// behavior is useful, for example, to make the text bold while using the
 /// default font family and size.
 ///
+/// When the widget has some selection, it will keep itself from disposing via
+/// [AutomaticKeepAliveClientMixin.wantKeepAlive] in order to avoid losing the
+/// selection. This may have implications for performance and for preventing
+/// parent widgets from disposing, such as [Scrollable]s that may expect to be
```

Comment:
I think we should not mention Scrollable/PrimaryScrollController because it is more of implementation detail? I think we should just state there is performance implication and state the consequence if there is any

Reply:
alternatively, if it the consequence is too serious we could make it not keep alive in any case, and it is up to the developer to keep it alive if they want.

Reply:
I agree with @chunhtai's first comment here. First sentence - :100:. Adding a mention of focus might be nice, since that is what having some selection appears to mean from this change. :)
I think leaving it up to the developer would make it harder for the developer to figure out what to do, especially if they try it themselves and then run into this weird crash. It seems like a really subtle thing that should just work.

Reply:
Thanks, I've simplified the wording a bit.

I'll keep an eye out in case anyone else mentions wanting to be able to control wantKeepAlive on SelectableText and EditableText.  Maybe we could add a parameter or something.  For now I think basing it on focus is the most intuitive thing in most cases.

Reply:
Sorry to metion here, but I just found out this is causing the bug for PageView as https://github.com/flutter/flutter/issues/93224 said. I think issue should  be reopened and should be noticed, It cost me a whole day to find out that `SelectableText` can mess up with `PageView`... 😞